packages/main/src/TimePickerBase.ts

```diff
@@ -75,16 +75,34 @@ import ResponsivePopoverCommonCss from "./generated/themes/ResponsivePopoverComm
 *
 * @event sap.ui.webc.main.TimePickerBase#change
  * @public
- */
-@event("change")
+ * @param {boolean} valid Indicator if the value is in correct format pattern and in valid range.
+*/
+@event("change", {
+	detail: {
+		valid: {
+			type: Boolean,
+		},
+	},
+})
 
 /**
  * Fired when the value of the <code>ui5-time-picker</code> is changed at each key stroke.
  *
  * @event sap.ui.webc.main.TimePickerBase#input
  * @public
- */
-@event("input")
+ * @param {string} value The current value.
+ * @param {boolean} valid Indicator if the value is in correct format pattern and in valid range.
+*/
+@event("input", {
+	detail: {
+		value: {
```

Comment:
if we remove it for the `change`, I think it should be removed from the `input` as well.

Reply:
I think that they will be different as input is the current while the value property will be the last submitted.

Reply:
Yes they are

Reply:
I see, if for the TimePicker's `input` event you say value is different and needs to be provided and we can't remove it, I am ok to stay for the `change` as well (as it was previously done in the initial commit).

I also saw that the DatePicker has `value` described for both the events, so it won't be a precedence.

On top, I checked the related issue,  the author mentioned to event details `value` and `valid`, in the context of DatePicker, so I will try to understand what the author is currently using for the TimePicker. In case they already use both `value` and `valid`, then we will know what to describe, what could avoid.

So, let's sync briefly tomorrow and will come up with the final decision.

Reply:
Discussed offline and decided to provide both `valid` and `value` as event details as in the DatePicker.